### PR TITLE
Weapon/Magazine Caliber Fix

### PR DESCRIPTION
**Pull request by InterroLouis**
## About The Pull Request

Fixes the issue of M42A rounds ejected from the rifle being unable to be put back into the magazine. 
## Why It's Good For The Game

It fixes a bug, as outlined in #2194

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -17,7 +17,7 @@
 	icon_state = "m42a"
 	item_state = "m42a"
 	max_shells = 15 //codex
-	caliber = "10x28mm Caseless" //codex
+	caliber = "10x28mm" //codex
```
Comment (DominikPanic) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: This is not a proper fix as other guns are likely to have the same issues, a codex variable shouldn't affect regular gameplay.

Comment (spookydonut) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: this isnt a codex variable the comment is wrong

Comment (DominikPanic) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Then the comment should be removed. Either way is fine, but right now the PR is not okay.

**Comment by DominikPanic** [comment]
Could you add "Fixes: #2194" in the description too? Otherwise it won't get linked via git. The title should be the name of the fix rather than the bug report, since that doesn't really tell anyone viewing it what the PR does.

**Comment by psykzz** [comment]
Should the magazine get updated to be the same caliber instead?
This means the sniper is going to start sharing ammo with a lot of other types of weapons, like sentries as well.

Im not sure if caseless was intentional for the caliber or not.

**Comment by InterroLouis** [comment]
You can't actually remove bullets from sentry magazines, or put them into it. So mechanically, it wouldn't make any difference. So the real question is, do we want the M42A to be caseless or not.

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -206,7 +206,7 @@
 	icon_state = "m42c"
 	item_state = "m42c"
 	max_shells = 6 //codex
-	caliber = "10x99mm Caseless" //codex
+	caliber = "10x99mm" //codex
```
Comment (DominikPanic) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]:
```suggestion
	caliber = "10x99mm"
```

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm (hunk 1 above)
Comment (DominikPanic) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]:
```suggestion
	caliber = "10x28mm"
```

Comment (Rohesie) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: The vars having a comment pointing they are codex-related is not a bad thing IMO.

Comment (DominikPanic) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: It is misleading in making it seems like they're ONLY there for codex.

Comment (Rohesie) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Ah, fair.

Comment (DominikPanic) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Either removing the comment or making it more clear is fine, but right now it's not ideal.

**Pull request merged**